rlclientlib/serialization/payload_serializer.cc

```diff
@@ -0,0 +1,16 @@
+#include "payload_serializer.h"
+
+namespace reinforcement_learning {
+  using namespace messages::flatbuff;
+  namespace logger {
+    v2::LearningModeType GetLearningMode(learning_mode mode) {
+      v2::LearningModeType result;
+      switch (mode) {
+      case APPRENTICE: return v2::LearningModeType_Apprentice;
+      case ONLINE: return v2::LearningModeType_Online;
+      case LOGGINGONLY: return v2::LearningModeType_LoggingOnly;
+      default: return v2::LearningModeType_Online;
```

Comment:
This function smells bad. Why is it not a cast and why the default case not an error?

Reply:
fixing default case handling. Regarding cast, not sure if it is a good idea - these are 2 independent enums defined in different places - casting them to each other can introduce hidden bugs if they become inconsistent at some point.

Reply:
Good point, if we can't eliminate one of them, a numeric cast is fragile long term.